tests/PHPStan/Command/ErrorFormatter/CheckstyleErrorFormatterTest.php

```diff
@@ -0,0 +1,121 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Command\ErrorFormatter;
+
+use PHPStan\Analyser\Error;
+use PHPStan\Command\AnalysisResult;
+use PHPStan\Command\ErrorsConsoleStyle;
+use PHPStan\TestCase;
+use Symfony\Component\Console\Input\StringInput;
+use Symfony\Component\Console\Output\Output;
+
+class CheckstyleErrorFormatterTest extends TestCase
+{
+
+	/**
+	 * Set up method
+	 *
+	 * @return void
+	 */
+	public function setUp()
+	{
+		$this->Formatter = new CheckstyleErrorFormatter();
+		$this->InputMock = $this->createMock(StringInput::class);
+	}
+
+	/**
+	 * Test FormatErrors method
+	 *
+	 * @return void
+	 */
+	public function testFormatErrors()
+	{
+		$analysisResultMock = $this->createMock(AnalysisResult::class);
+		$analysisResultMock
+			->expects($this->at(0))
+			->method('hasErrors')
+			->willReturn(true);
+
+		$analysisResultMock
+			->expects($this->at(1))
+			->method('getFileSpecificErrors')
+			->willReturn([
+				new Error('Foo', 'foo.php', 1),
+				new Error('Bar', 'bar.php', 2),
+			]);
+
+		$output = new TestOutput();
+		$style  = new ErrorsConsoleStyle($this->InputMock, $output);
+
+		$this->assertEquals(1, $this->Formatter->formatErrors($analysisResultMock, $style));
+		$this->assertEquals('<?xml version="1.0" encoding="UTF-8"?>
+<checkstyle>
+<file name="foo.php">
+ <error line="1" column="1" severity="error" message="Foo"/>
+</file>
+<file name="bar.php">
+ <error line="2" column="1" severity="error" message="Bar"/>
+</file>
+</checkstyle>
+', $output->output);
+	}
+
+	/**
+	 * Test FormatErrors method
+	 *
+	 * @return void
+	 */
+	public function testFormatErrorsEmpty()
+	{
+		$analysisResultMock = $this->createMock(AnalysisResult::class);
+		$analysisResultMock
+			->expects($this->at(0))
+			->method('hasErrors')
+			->willReturn(false);
+
+		$output = new TestOutput();
+		$style  = new ErrorsConsoleStyle($this->InputMock, $output);
+
+		$this->assertEquals(0, $this->Formatter->formatErrors($analysisResultMock, $style));
+		$this->assertEquals('<?xml version="1.0" encoding="UTF-8"?>
+<checkstyle>
+</checkstyle>
+', $output->output);
+	}
+
+}
+
+/**
+ * Class for testing console output
+ */
+class TestOutput extends Output
```

Comment:
I've tested command output like this, so please use the same approach: https://github.com/phpstan/phpstan/blob/master/tests/PHPStan/Command/AnalyseApplicationIntegrationTest.php#L50